### PR TITLE
Fix | Send bulk data as unicode when destination column is nchar/nvarchar

### PR DESCRIPTION
**Pull request by ulvii**
#1181

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java [hunk 1]
```diff
@@ -3528,4 +3561,16 @@ protected void setStmtColumnEncriptionSetting(
...
+    /**
+     * For the case when source database stores unicode data in CHAR/VARCHAR and destination column is NCHAR/NVARCHAR.
+     *
+     * @param jdbcType
+     * @param ssType
+     * @return whether conversion to unicode is required.
+     */
+    private boolean unicodeConversionRequired(int jdbcType, SSType ssType) {
+        return ((java.sql.Types.CHAR == jdbcType || java.sql.Types.VARCHAR == jdbcType
+                || java.sql.Types.LONGNVARCHAR == jdbcType) && (SSType.NCHAR == ssType || SSType.NVARCHAR == ssType));
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java]: Shouldn't we also check for SSType.NVARCHARMAX?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java [hunk 2]
```diff
@@ -2168,33 +2193,41 @@ else if (null != sourceCryptoMeta) {
...
+                                    tdsWriter.writeBytes(bytes);
+                                } else {
+                                    tdsWriter.writeShort((short) (colValueStr.length()));
+                                    // converting string into destination collation using Charset
 
-                                SQLCollation destCollation = destColumnMetadata.get(destColOrdinal).collation;
-                                if (null != destCollation) {
-                                    tdsWriter.writeBytes(colValueStr
-                                            .getBytes(destColumnMetadata.get(destColOrdinal).collation.getCharset()));
+                                    SQLCollation destCollation = destColumnMetadata.get(destColOrdinal).collation;
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java]: ? is that a tab?

File: src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java [hunk 3]
```diff
@@ -36,12 +36,12 @@
...
+     * @throws SQLException
      */
     @Test
-    public void testDriverDM() throws ClassNotFoundException {
-        Stream<Driver> drivers = DriverManager.drivers();
-        Object[] driversArray = drivers.toArray();
-        assertEquals(driversArray[0].getClass(), Class.forName(Constants.MSSQL_JDBC_PACKAGE + ".SQLServerDriver"),
+    public void testDriverDM() throws SQLException, ClassNotFoundException {
+        Driver driver = DriverManager.getDriver(connectionString);
+        assertEquals(driver.getClass(), Class.forName(Constants.MSSQL_JDBC_PACKAGE + ".SQLServerDriver"),
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/SQLServerDriverTest.java]: expected is 1st param

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java (hunk 2 above)
Comment (ulvii) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java]: No it's not, the file is formatted.

**Pull request merged**